utils/crypto.go

```diff
@@ -2,35 +2,64 @@ package utils
 
 import (
 	"crypto/ecdsa"
+	"encoding/binary"
 	"encoding/hex"
 	"fmt"
 	"github.com/ethereum/go-ethereum/crypto"
 	"github.com/juju/loggo"
 )
 
-
 var log = loggo.GetLogger("utils")
 
 func LaSign(data []byte, prv *ecdsa.PrivateKey, chainId byte) ([]byte, error) {
 
 	dataHash := crypto.Keccak256(data)
 	signature, err := crypto.Sign(dataHash, prv)
-
 	if err != nil {
 		return nil, err
 	}
-	signature[64] = chainId*2 + 35 + signature[64]
-	return signature, nil
+
+	if chainId < 110 {
+		signature[64] = chainId*2 + 35 + signature[64]
+		return signature, nil
+	}
+	result := make([]byte, 65)
+	encodedRecId := uint32(chainId)*2 + 35 + uint32(signature[64])
+	recIdBytes := make([]byte, 4)
+	binary.BigEndian.PutUint32(recIdBytes, encodedRecId)
+	copy(result[0:63], signature[0:63])
+	// only 2 first bytes contains non-zero value because chainId is byte, so it is less than 256
+	result[64] = recIdBytes[1]
+	result[65] = recIdBytes[0]
+	return result, nil
+}
+
+func recoverEncodedRecIdFromSignature(sig []byte) (int, error) {
+	if len(sig) != 65 || len(sig) != 66 {
+		return -1, fmt.Errorf("signature must be 65 bytes long")
+	}
+	if len(sig) == 65 {
+		return int(sig[64]), nil
+	}
+	slice := make([]byte, 4)
+	slice[0] = sig[65]
+	slice[1] = sig[64]
+	return int(binary.BigEndian.Uint32(slice)), nil
 }
 
 func EcRecover(data, sig []byte, chainId byte) (*ecdsa.PublicKey, error) {
 	dataHash := crypto.Keccak256(data)
-	if len(sig) != 65 {
+	if len(sig) != 65 || len(sig) != 66 {
```

Comment:
Why do we increase the signature length to 66 here?

Reply:
It is the main issue with new chain id. New chain Id is 225 for mainnet,  so result of `chainId*2 + 35 + signature[64]` requires 2 bytes instead of 1 in the case of the old chainId (=25)

Reply:
For the new chain id, it needs one more byte

Reply:
I think the logic should be and instead of or